Classes/Outlet/OutletArgument.php

```diff
@@ -0,0 +1,243 @@
+<?php
+namespace FluidTYPO3\Flux\Outlet;
+
+/*
+ * This file is part of the FluidTYPO3/Flux project under GPLv2 or later.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.md file that was distributed with this source code.
+ */
+
+
+use TYPO3\CMS\Extbase\Mvc\Controller\MvcPropertyMappingConfiguration;
+use TYPO3\CMS\Extbase\Property\Exception\TargetNotFoundException;
+use TYPO3\CMS\Extbase\Utility\TypeHandlingUtility;
+use TYPO3\CMS\Extbase\Validation\Exception\NoSuchValidatorException;
+use TYPO3\CMS\Extbase\Validation\Validator\ConjunctionValidator;
+use TYPO3\CMS\Extbase\Validation\Validator\ValidatorInterface;
+use TYPO3\CMS\Extbase\Validation\ValidatorResolver;
+
+/**
+ * A outlet argument
+ *
+ * @api
+ */
+class OutletArgument
+{
+    /**
+     * @var \TYPO3\CMS\Extbase\Property\PropertyMapper
+     */
+    protected $propertyMapper;
+
+    /**
+     * @var MvcPropertyMappingConfiguration
+     */
+    protected $propertyMappingConfiguration;
+
+    /**
+     * @var \TYPO3\CMS\Extbase\Validation\ValidatorResolver
+     * @inject
+     */
+    protected $validatorResolver;
+
+    /**
+     * Name of this argument
+     *
+     * @var string
+     */
+    protected $name = '';
+
+    /**
+     * Data type of this argument's value
+     *
+     * @var string
+     */
+    protected $dataType = null;
+
+    /**
+     * Actual value of this argument
+     *
+     * @var mixed
+     */
+    protected $value = null;
+
+    /**
+     * A custom validator, used supplementary to the base validation
+     *
+     * @var array<\TYPO3\CMS\Extbase\Validation\Validator\ValidatorInterface>
+     */
+    protected $validators = null;
+
+    /**
+     * The validation results. This can be asked if the argument has errors.
+     *
+     * @var \TYPO3\CMS\Extbase\Error\Result
+     */
+    protected $validationResults = null;
+
+    /**
+     * @param \TYPO3\CMS\Extbase\Property\PropertyMapper $propertyMapper
+     */
+    public function injectPropertyMapper(\TYPO3\CMS\Extbase\Property\PropertyMapper $propertyMapper)
+    {
+        $this->propertyMapper = $propertyMapper;
+    }
+
+    /**
+     * @param \TYPO3\CMS\Extbase\Mvc\Controller\MvcPropertyMappingConfiguration $propertyMappingConfiguration
+     */
+    public function injectPropertyMappingConfiguration(MvcPropertyMappingConfiguration $propertyMappingConfiguration)
+    {
+        $this->propertyMappingConfiguration = $propertyMappingConfiguration;
+    }
+
+    /**
+     * @param \TYPO3\CMS\Extbase\Validation\ValidatorResolver $validatorResolver
+     */
+    public function injectValidatorResolver(ValidatorResolver $validatorResolver)
+    {
+        $this->validatorResolver = $validatorResolver;
+    }
+
+    /**
+     * Constructs this controller argument
```

Comment:
Please be more specific or omit. Which controllers' argument is being constructed (by a constructor ^^)? :)